triggermail_templates.py

```diff
@@ -128,6 +134,53 @@ def dissect_filename(self, template_filename):
         self.partner = self.settings.get("partner", self.partner) or self.partner
         self.partner = self.partner.replace("_templates", "")
 
+    def generate_pertinent_file_names_and_map(self, template_filename):
+        template_filename = template_filename.replace(self.path + '/', '')
+        image_regex = "/img/" + self.partner + "/" + "(.*?)['\"]"
+        file_map = dict()
+        all_file_names_required = [template_filename]
+        all_images_required = []
+        # go through all files this template requires
+        # we get all the filenames, their contents, and images
+        for file_name in all_file_names_required:
+            # some templates include html files that don't exist. Just because they don't exist, doesn't
+            # mean it should automatically fail the command, they are most likely in jinja "if" statements
+            try:
+                contents = read_file(os.path.join(self.path, file_name))
+            except:
+                continue
+            file_map[file_name] = contents
+            # this regex finds all the other files the html file depends on. Because we're thenadding to
+            # the loop we're cycling through, this can be recursive
+            files_required_by_file = [file_name for _,file_name in \
+                                        re.findall("{%\s*(include|extends|import)\s*['\"](.*?)['\"]", contents)]
+            all_file_names_required += [x for x in files_required_by_file if x not in all_file_names_required]
+            images_required_by_file = re.findall(image_regex, contents)
+            all_images_required += [img for img in re.findall(image_regex, contents) if img not in all_images_required]
```

Comment:
It would theoretically be better for all_images_required to be a `set()`, since testing if something is in a set is constant time. In fact, this could then become:

`all_images_require.update(re.findall(image_regex, contents))` (I think)